tests/test_tempfile.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+
+import pytest
+
+from anyio import (
+    NamedTemporaryFile,
+    SpooledTemporaryFile,
+    TemporaryDirectory,
+    TemporaryFile,
+    gettempdir,
+    gettempdirb,
+    gettempprefix,
+    gettempprefixb,
+    mkdtemp,
+    mkstemp,
+    to_thread,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+@pytest.mark.anyio
+async def test_temporary_file() -> None:
+    data = b"temporary file data"
+    async with TemporaryFile[bytes]() as af:
+        await af.write(data)
+        await af.seek(0)
+        result = await af.read()
+    assert result == data
+    assert af.closed
+
+
+@pytest.mark.anyio
+async def test_named_temporary_file() -> None:
+    data = b"named temporary file data"
+    async with NamedTemporaryFile[bytes]() as af:
+        filename: str = str(af.name)
+        assert os.path.exists(filename)
+        await af.write(data)
+        await af.seek(0)
+        result = await af.read()
+    assert result == data
+    assert not os.path.exists(filename)
+
+
+@pytest.mark.anyio
+async def test_spooled_temporary_file_io_and_rollover() -> None:
+    data = b"spooled temporary file data" * 3
+    async with SpooledTemporaryFile[bytes](max_size=10) as stf:
+        await stf.write(data)
+        await stf.seek(0)
+        result = await stf.read()
+        assert result == data
+        pos = await stf.tell()
+        assert isinstance(pos, int)
+        await stf.rollover()
+        assert not stf.closed
+    assert stf.closed
+
+
+@pytest.mark.anyio
+async def test_spooled_temporary_file_error_conditions() -> None:
+    stf = SpooledTemporaryFile[bytes]()
+    with pytest.raises(RuntimeError):
+        await stf.rollover()
+    with pytest.raises(AttributeError):
+        _ = stf.nonexistent_attribute
+
+
+@pytest.mark.anyio
+async def test_temporary_directory_context_manager() -> None:
+    async with TemporaryDirectory() as td:
+        td_path = pathlib.Path(td)
+        assert td_path.exists() and td_path.is_dir()
+        file_path = td_path / "test.txt"
+        file_path.write_text("temp dir test", encoding="utf-8")
+        assert file_path.exists()
+    assert not td_path.exists()
+
+
+@pytest.mark.anyio
+async def test_temporary_directory_cleanup_method() -> None:
+    td = TemporaryDirectory()
+    td_str = await td.__aenter__()
+    td_path = pathlib.Path(td_str)
+    file_path = td_path / "file.txt"
+    file_path.write_text("cleanup test", encoding="utf-8")
+    await td.cleanup()
+    assert not td_path.exists()
+
+
+@pytest.mark.anyio
+async def test_mkstemp() -> None:
+    fd, path_ = await mkstemp(suffix=".txt", prefix="mkstemp_", text=True)
+    assert isinstance(fd, int)
+    assert isinstance(path_, str)
```

Comment:
Unless I'm missing something, there is no reason to add the underscore suffix to `path`.
```suggestion
    fd, path = await mkstemp(suffix=".txt", prefix="mkstemp_", text=True)
    assert isinstance(fd, int)
    assert isinstance(path, str)
```